src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsics.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.Maintainability
+{
+    using static MicrosoftCodeQualityAnalyzersResources;
+
+    /// <summary>
+    /// CA1516: <inheritdoc cref="UseCrossPlatformIntrinsicsTitle"/>
+    /// </summary>
+    public abstract class UseCrossPlatformIntrinsicsAnalyzer : DiagnosticAnalyzer
```

Comment:
Class name ending with `Analyzer` but file name isn't. I would expect file name to match, unless this is a pattern taken already in the repo.